controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -128,21 +135,18 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
         Pair<Byte, Integer> extracted = extractFromPolicy(policy);
 
         WireCommands.CreateSegment request = new WireCommands.CreateSegment(requestId, qualifiedStreamSegmentName,
-                extracted.getLeft(), extracted.getRight(), controllerToken);
-        sendRequestAsync(request, replyProcessor, result, clientCF, ModelHelper.encode(uri));
+                extracted.getLeft(), extracted.getRight(), retrieveDelegationToken());
```

Comment:
@tkaitchuck 
> Here and below this `retrieveDelegationToken` works quite differently than it did before. Rather than being passed where needed a token is created on demand. 

The logic has not been changed in any way. Earlier this same thing was done in `StreamMetadataTasks.java` and it has been moved inside segment helper now. 
All controller to segment store interactions have always used this master token 

> This token is also not just a delegation token associated with the origin of the request it is a new master token. This represents a substantive change in the way tokens work and potentially weakens them.

Prior to pravega tables, all controller-segment store interactions have been invoked typically asynchronously via some background processing so the origin's token is lost. So this PR is not changing anything in how things happened wrt auth earlier. 

> I don't think we want to be sending a new master token in every instance. If that is the case here, then it should be clearly spelled out in an issue.

I agree, we should reuse the token as long as it is valid (this wasnt the case earlier) and i will take that up as a separate issue.